lib/main/timing/TimingCollector.ts

```diff
@@ -0,0 +1,305 @@
+import { getCurrentUserId } from '../store'
+import { platform, hostname, arch } from 'os'
+import { performance } from 'perf_hooks'
+import { analytics } from '@/app/components/analytics'
+import { app } from 'electron'
+import { TimingReport, TimingEvent } from '@/app/generated/ito_pb'
+import { grpcClient } from '../../clients/grpcClient'
+import { interactionManager } from '../interactions/InteractionManager'
+
+/**
+ * Enum for all tracked timing events in the interaction lifecycle
+ */
+export enum TimingEventName {
+  // Core interaction events
+  INTERACTION_ACTIVE = 'interaction_active',
+
+  // Server communication
+  SERVER_TRANSCRIBE = 'server_transcribe',
+
+  // Context and processing
+  CONTEXT_GATHER = 'context_gather',
```

Comment:
This might be too generic -- this is solely limited to getting the highlighted text, right? Maybe scope it down to that